### DISCUSSION TITLE
Rendering CSV with Additional Fields

### DISCUSSION THREAD

**Discussion by fiftyghosts** [kernc/backtesting.py#848] [Q&A]
I'm trying to add the stop loss price and tp profit price to the CSV rendered during this call:

`stats['_trades'].to_csv(csv_filename)`

I know they were separate orders, I need those values for a stats db I'm building. I want those values whether they were hit, or not. 

I'm currently trying to alter the `_close_trade()` method in `backtesting.py` file where the `self.closed_trades.append()` is called. I've been able to get the fields to render in the CSV file, but not any values.

Any ideas?

Thank you.

**Comment by kernc** [comment]
You could try adding extra columns "sl" and "tp" here:
https://github.com/kernc/backtesting.py/blob/8c13d005a2c07ce6bb5d6c331a891f72948d723b/backtesting/_stats.py#L58-L69
Please report your findings and whether that should be added.

**Reply by fiftyghosts** [reply]
Hi Kernc, 

Thank you for your response.  That was one of the first things I tried, but no data was being populated, so nothing showed up in the CSV except the empty fields. Well, one did, but that's it. I think that one item showed up because it was an order the actually executed. What I'm looking for is to know what the stop loss was for a given "set" of orders. In other words: a trade is entered with a stop loss order set, I want to get the usual trade data plus what the stop loss _would_ have been, had it been triggered.

```
'StopLoss': [t.sl for t in trades],
 'TakeProfit': [t.tp for t in trades]
```

Any ideas?

**Reply by kernc** [reply]
See https://github.com/kernc/backtesting.py/issues/56.

**Comment by fiftyghosts** [comment]
Thank you for your response!

I've moved to another task, but this is in my queue.  I'll let you know
what happens.
